src/api/app/controllers/webui/request_controller.rb

```diff
@@ -7,7 +7,7 @@ class Webui::RequestController < Webui::WebuiController
 
   before_action :require_login,
                 except: %i[show beta_show sourcediff diff request_action request_action_changes request_action_details inline_comment build_results
-                           changes changes_diff mentioned_issues]
+                           changes mentioned_issues]
```

Comment:
This will solve the issue, yes, for now.

However, as of now, unauthenticated users can still see the diff of the changes using the old request view (not using the `request_show_redesign` feature flag), if I'm not mistaken.

With the suggested changes applied, once the Request Redesign opportunity leaves beta and we no longer use the `request_show_redesign` feature flag, unauthenticated users will no longer be able to see the diff of the changes (the content rendered by the `changes_diff` action).

If we decide to go down this path, I would suggest at least adding a comment, either in the codebase, the pull request, or the commit message.

Reply:
I agree, that’s a valid point. Ideally, it should stay under `require_login` for now. Since this is only temporary until the feature is fully rolled out, I don’t think we need to address this issue and close this PR